src/workerd/api/sql-test.js

```diff
@@ -60,6 +61,67 @@ async function test(storage) {
     assert.equal(result[2]['value'], 3)
   }
 
+  // Test multiple statements in .exec
+  {
+    const result = Array.from(sql.exec(`SELECT 123; SELECT 456;`))
+    assert.deepEqual(result, [{ 456: 456 }])
+  }
+  {
+    const result = Array.from(
+      sql.exec(`SELECT 123; SELECT 456; -- trailing comment`)
+    )
+    assert.deepEqual(result, [{ 456: 456 }])
+  }
+  {
+    const result = Array.from(
+      sql.exec(
+        `SELECT 123; SELECT 456; -- trailing comment\n-- with whitespace\n `
+      )
+    )
+    assert.deepEqual(result, [{ 456: 456 }])
+  }
+  {
+    const result = Array.from(
+      sql.exec(`SELECT 123; --middle comment\n SELECT 456;`)
+    )
+    assert.deepEqual(result, [{ 456: 456 }])
+  }
+  {
+    const result = Array.from(
+      sql.exec(`-- leading comment\n;SELECT 123; SELECT 456;`)
+    )
+    assert.deepEqual(result, [{ 456: 456 }])
+  }
+  assert.throws(() => sql.exec('\n'), 'SQL code did not contain a statement')
+  assert.throws(
+    () => sql.exec('\n-- comment'),
+    'SQL code did not contain a statement'
+  )
+
+  // Test partial query ingestion
+  assert.deepEqual(
+    sql.ingest(`SELECT 123; SELECT 456; -- trailing comment`),
+    23
+  )
+  assert.deepEqual(sql.ingest(`SELECT 123; SELECT 456;    `), 23)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELECT 456;`), 23)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELECT 456`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELECT 45`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELECT 4`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELECT `), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELECT`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELEC`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SELE`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SEL`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; SE`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; S`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123; `), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123;`), 11)
+  assert.deepEqual(sql.ingest(`SELECT 123`), 0)
+  assert.deepEqual(sql.ingest(`SELECT 12`), 0)
+  assert.deepEqual(sql.ingest(`SELECT 1`), 0)
```

Comment:
It would probably be good to have a test case for just a semicolon (";") and for something where there's a semicolon inside another part of the statement (e.g., a string) just as a confidence test.

Reply:
Will do, although with my latest refactoring `';'` will fail, and so will `SELECT 123;; SELECT 456;` because `prepareSql` explodes on all empty statements (which for things like trailing comments has been raised as a bug in the past).

Will fix and push test cases for handling semicolons/comments before merging this PR